src/crafting.cpp

```diff
@@ -432,6 +432,31 @@ std::list<item> player::consume_components_for_craft( const recipe &making, int
     return used;
 }
 
+std::list<item> player::consume_some_components_for_craft( const recipe &making, int batch_size )
+{
+    std::list<item> used;
+    if( has_trait( trait_id( "DEBUG_HS" ) ) ) {
+        return used;
+    }
+    const auto &req = making.requirements();
+    int cou = 0;
+    for( const auto &it : req.get_components() ) {
+        // Each component currently has 50% chance of not being consumed
+        int chc = 1;
+        int ran = rng( 0, chc );
+        // Skip first item so failed craft with one item recipe always loses component
+        if( cou > 0 && ran == chc ) {
```

Comment:
The `chc` and `ran` variables ain't really needed. You can remove them altogether and replace the check with 
```suggestion
        if( cou > 0 && one_in( 2 ) ) {
```

Reply:
@Night-Pryanik thanks! I'll get another PR submitted to clean that up.